aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -904,19 +904,11 @@ Tensor mvlgamma(const Tensor& self, int64_t p) {
   return args.lgamma_().sum(-1).add_(p2_sub_p * std::log(c10::pi<double>) * QUARTER);
 }
 
+// since mvlgamma_ has different signature from its
+// out and functional variant, we explicitly
+// define it (instead of using structured kernel).
 Tensor& mvlgamma_(Tensor& self, int64_t p) {
-  mvlgamma_check(self, p);
-  Tensor args = native::arange(
-      -p *HALF  + HALF,
-      HALF,
-      HALF,
-      optTypeMetaToScalarType(self.options().dtype_opt()),
-      self.options().layout_opt(),
-      self.options().device_opt(),
-      self.options().pinned_memory_opt());
-  args = args.add(self.unsqueeze(-1));
-  const auto p2_sub_p = static_cast<double>(p * (p - 1));
-  return self.copy_(args.lgamma_().sum(-1).add_(p2_sub_p * std::log(c10::pi<double>) * QUARTER));
+  return at::mvlgamma_out(self, self, p);
```

Comment:
Do you know why the argument order is flipped compared to the function just below?

Reply:
Thanks for catching it @albanD. You're absolutely right, the argument order was incorrect. The issue was that I had
```cpp
return at::mvlgamma_out(self, self, p); 
```

according to the generated signature in the header
```
 at::Tensor & mvlgamma_out(const at::Tensor & self, int64_t p, at::Tensor & out);
```

It should be:
```
return at::mvlgamma_out(self, p, self);  
```
so now this matches the schema in `native_functions.yaml` and the pattern used by `polygamma_` just above. I've fixed this and verified all tests still pass. Thanks for catching this!